libkbfs/folder_block_ops.go

```diff
@@ -1247,14 +1284,34 @@ func (fbo *folderBlockOps) getOrCreateSyncInfoLocked(
 	return si, nil
 }
 
-// GetDirtyRefs returns a list of references of all known dirty
-// blocks.
-func (fbo *folderBlockOps) GetDirtyRefs(lState *lockState) []BlockRef {
+// GetDirtyFiles returns a list of references of all known dirty
+// files.
+func (fbo *folderBlockOps) GetDirtyFiles(lState *lockState) []BlockRef {
 	fbo.blockLock.RLock(lState)
 	defer fbo.blockLock.RUnlock(lState)
 	var dirtyRefs []BlockRef
 	for ref := range fbo.deCache {
-		dirtyRefs = append(dirtyRefs, ref)
+		if _, ok := fbo.unrefCache[ref]; ok {
```

Comment:
Good question -- just checking whether it's a file type is not sufficient, because there may be file entries that don't need syncing.  For example, a file whose mtime is set needs to have a cached `DirEntry`, but it doesn't count as a "dirty file" because it hasn't been written to.